codegenerator/cli/templates/static/codegen/src/eventFetching/SourceManager.res

```diff
@@ -0,0 +1,176 @@
+open Belt
+
+let toQueryId = (query: FetchState.nextQuery) => {
+  query.fetchStateRegisterId->FetchState.registerIdToString ++ "-" ++ query.fromBlock->Int.toString
+}
+
+// Can't simply store fetching partitions, since fetchBatch
+// can be called with old chainFetchers after isFetching was set to false,
+// but state isn't still updated with fetched data.
+// This is temporary until all fetching logic moved to mutable state,
+// but for now prevent fetching the same partition twice with the same query,
+// using lastFetchedQueryId (aka idempotency key)
+type partitionFetchingState = {
+  isFetching: bool,
+  lastFetchedQueryId?: string,
+}
```

Comment:
We'll be able to get rid of these, after the code start managing fetching itself instead of being called by global state module.